ngraph/core/reference/include/ngraph/runtime/reference/convolution.hpp

```diff
@@ -24,338 +25,219 @@
+                                {
+                                    for (int f_z = 0; f_z < filter_size_z; ++f_z)
+                                    {
+                                        for (int f_y = 0; f_y < filter_size_y; ++f_y)
+                                        {
+                                            for (int f_x = 0; f_x < filter_size_x; ++f_x)
+                                            {
+                                                int rel_i_z = i_z + (f_z * p.dilation[0]);
+                                                int rel_i_y = i_y + (f_y * p.dilation[1]);
+                                                int rel_i_x = i_x + (f_x * p.dilation[2]);
+
+                                                bool padding =
+                                                    !(in_range(rel_i_x, {0, input_size_x}) &&
+                                                      in_range(rel_i_y, {0, input_size_y}) &&
+                                                      in_range(rel_i_z, {0, input_size_z}));
+                                                if (padding)
+                                                    continue;
+
+                                                int f_buf_idx =
+                                                    (f_z * filter_size_y * filter_size_x) +
+                                                    (f_y * filter_size_x) + f_x;
+                                                int i_buf_idx =
+                                                    (rel_i_z * input_size_y * input_size_x) +
+                                                    (rel_i_y * input_size_x) + rel_i_x;
+                                                sum += static_cast<ACCU>(input_channel[i_buf_idx]) *
+                                                       static_cast<ACCU>(filter_channel[f_buf_idx]);
+                                            }
+                                        }
+                                    }
+                                    input_channel += input_channel_size;
+                                    filter_channel += filter_channel_size;
+                                }
+                                *out = sum;
+                                ++out;
                             }
                         }
-                        ++in_it;
-                        ++filter_it;
                     }
+                }
 
-                    out[out_transform.index(out_coord)] = result;
+                void extend_to_3D(ConvolutionParams& p, Shape& in_shape, Shape& filter_shape)
+                {
+                    int spatial_rank = in_shape.size() - 2;
+                    if (spatial_rank < 3)
+                    {
+                        int missing_dims = 3 - spatial_rank;
+                        p.dilation.insert(
+                            std::next(p.dilation.end(), -spatial_rank), missing_dims, 1);
```

Comment:
Consider `std::next` -> `std::prev`.

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/3922/commits/4761ecd06af58f88cd2b9a4767392ba6dd8f0a8f).